### PR TITLE
improve run script

### PR DESCRIPTION
**Pull request by ceyonur**
we can omit downloading & extracting steps if we already did them. also waits until sigterm received in run mode so we can shutdown background processes.

### CODE REVIEW

File: scripts/run.sh [hunk 1]
```diff
@@ -19,14 +19,14 @@ fi
 VERSION=$1
 if [[ -z "${VERSION}" ]]; then
   echo "Missing version argument!"
-  echo "Usage: ${0} [VERSION] [GENESIS_ADDRESS]" >> /dev/stderr
+  echo "Usage: ${0} [VERSION] [GENESIS_ADDRESS]" >>/dev/stderr
```
Comment (aaronbuchwald) [inline scripts/run.sh:22]: Is there a reason for the removed spaces? If not, can we add them back in?

File: scripts/run.sh [hunk 2]
```diff
@@ -49,45 +49,48 @@ echo AVALANCHE_LOG_LEVEL: ${AVALANCHE_LOG_LEVEL}
 # https://github.com/ava-labs/avalanchego/releases
 GOARCH=$(go env GOARCH)
 GOOS=$(go env GOOS)
-DOWNLOAD_URL=https://github.com/ava-labs/avalanchego/releases/download/v${VERSION}/avalanchego-linux-${GOARCH}-v${VERSION}.tar.gz
-DOWNLOAD_PATH=/tmp/avalanchego.tar.gz
+BASEFILE=/tmp/subnet-evm-runner
```
Comment (aaronbuchwald) [inline scripts/run.sh:52]: Let's discuss switching to a standard approach of how we install avalanchego across different repos. We have a similar script in a few places now and maintaining them is getting difficult and fractured.

File: scripts/run.sh (hunk 1 above)
Comment (ceyonur) [reply scripts/run.sh:22]: no, seems my editor extension formatted it.

File: scripts/run.sh [hunk 3]
```diff
@@ -257,16 +261,10 @@ if [[ ${MODE} == "test" ]]; then
...
-  echo "network-runner RPC server is running on PID ${PID}..."
-  echo ""
-  echo "use the following command to terminate:"
-  echo ""
-  echo "pkill -P ${PID}"
-  echo "kill -2 ${PID}"
-  echo "pkill -9 -f srEXiWaHuhNyGwPUi444Tu47ZEDwxTWrbQiuD7FmgSAQ6X7Dy"
-  echo ""
+  echo "CTRL + C to exit and kill all background processes"
+  wait
```
Comment (aaronbuchwald) [inline scripts/run.sh:269]: Isn't killing the processes still necessary with the ANR?

File: scripts/run.sh [hunk 4]
```diff
@@ -192,37 +195,38 @@ EOF
 # https://github.com/ava-labs/avalanche-network-runner
 # TODO: use "go install -v github.com/ava-labs/avalanche-network-runner/cmd/avalanche-network-runner@v${NETWORK_RUNNER_VERSION}"
 NETWORK_RUNNER_VERSION=1.0.12
```
Comment (aaronbuchwald) [inline scripts/run.sh:197]: Can we take in the network runner version from `scripts/versions.sh` and keep it up to date there?

Comment (ceyonur) [reply scripts/run.sh:197]: aaahhh I requested that it in #155 but seems forgot to actually send the review...  geez...
image

File: .github/workflows/lint-tests-release.yml [hunk 5]
```diff
@@ -57,13 +57,13 @@ jobs:
...
         with:
-          node-version: '14.x'
+          node-version: "14.x"
       - name: Yarn install
         run: yarn
         working-directory: ./contract-examples
       - name: Run e2e tests
         shell: bash
-        run: E2E=true scripts/run.sh 1.7.13 0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC
+        run: E2E=true AVALANCHE_VERSION=v1.7.13 scripts/run.sh
```
Comment (darioush) [inline .github/workflows/lint-tests-release.yml:66]: can we go the newest version here?

Comment (ceyonur) [reply .github/workflows/lint-tests-release.yml:66]: sure

File: scripts/run.sh [hunk 6]
```diff
@@ -52,45 +48,48 @@ echo AVALANCHE_LOG_LEVEL: ${AVALANCHE_LOG_LEVEL}
 # https://github.com/ava-labs/avalanchego/releases
 GOARCH=$(go env GOARCH)
 GOOS=$(go env GOOS)
-DOWNLOAD_URL=https://github.com/ava-labs/avalanchego/releases/download/v${VERSION}/avalanchego-linux-${GOARCH}-v${VERSION}.tar.gz
-DOWNLOAD_PATH=/tmp/avalanchego.tar.gz
+BASEFILE=/tmp/subnet-evm-runner
```
Comment (aaronbuchwald) [inline scripts/run.sh:51]: Nit: can we change `BASEFILE` to `BASEDIR` ?

**Review by aaronbuchwald — APPROVED** [review]
LGTM with one small nit to update a variable name.

File: scripts/run.sh (hunk 6 above)
Comment (ceyonur) [reply scripts/run.sh]: done

**Review by darioush — APPROVED** [review]

**Pull request merged**